### ISSUE TITLE
`woman-manpath` variable wrong on macOS/OpenBSD

### ISSUE THREAD

**Issue by citizen428** [doomemacs/doomemacs#7021]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27 or 28. *(Doom does not support development builds like 29+ or any version ending in .50 or .9x)*

### Expected behavior

The variable `woman-manpath` contains the correct paths for manpage lookup and `M-x woman` can be used to read man pages.

Note: since `woman` only correctly deals with `troff` formatted man pages and BSDs tend to use `mandoc` this isn't the biggest issue.

### Current behavior

Since `man --path` is not supported on macOS (and OpenBSD, presumably also not on other BSD flavors) the variable `woman-manpath` contains an error message instead of different paths:

```elisp
("/usr/bin/man" " illegal option -- -\nUsage" "\n man [-adho] [-t | -w] [-M manpath] [-P pager] [-S mansect]\n     [-m arch[" "machine]] [-p [eprtv]] [mansect] page [...]\n man -f page [...] -- Emulates whatis(1)\n man -k page [...] -- Emulates apropos(1)")
```

The source of the problem can be found in `~/.config/emacs/modules/config/default/config.el`:

```elisp
(after! woman
  ;; The woman-manpath default value does not necessarily match man. If we have
  ;; man available but aren't using it for performance reasons, we can extract
  ;; it's manpath.
  (when (executable-find "man")
    (setq woman-manpath
          (split-string (cdr (doom-call-process "man" "--path"))
                        path-separator t))))
```

Since the `--path` option for `man` isn't available on BSDs and `woman` isn't particularly useful there anyway, this could potentially be fixed by adding a check for the operating system:

```diff
-  (when (executable-find "man")
+  (when (and IS-LINUX (executable-find "man"))
```

I'm happy to make a PR if people are ok with this solution.

Unrelated: there's also a typo in the docstring, it should read "its manpath", not "it's manpath".

### Steps to reproduce

1. Open Emacs
2. Make sure `woman` is loaded (e.g. `M-x woman`)
3. Inspect the value of `woman-manpath` (`Spc-h v woman-manpath`)

### System Information

https://pastebin.com/kpfruSUH

**Comment by witty-pigeon** [comment]
I cannot create a PR right now since `modules` are in the [DNP](https://doomemacs.org/d/do-not-pr) list, but a fix is available in my branch.
Possibly I'll create a PR when the modules are transferred over to the new repo

**Comment by John-Goff** [comment]
Ran into this today and put together a quick PR before I found this issue. I do not believe that this is covered under the Do Not PR as it's a fix to a core module and not a part of the separate modules project as of yet, but if a maintainer wants to close the PR then they can. I also went the route of defaulting to `manpath` as opposed to checking the platform, as the Linux systems which I tried did support `manpath`. If there's any issues on a Linux system with `manpath` then perhaps checking the platform may be a better solution, but from what I could see this PR covers both.
